### PR TITLE
chore: Use "styles" and move them outside themes

### LINKED ISSUE #69: Better importing of Type Tokens
Right now, I import Type Tokens like so:

`import { headingLarge, headingMedium, headingSmall } from 'themes/mixins';`

...but want I want to do is like so:

`import { heading } from 'themes/typeTokens';`

...and then:

`heading.Large`

SHAVE ME A YAK PLS.

### PR DESCRIPTION
**Pull request by tofumatt**
Closes #69.

Let me know if this all makes sense. I've made things available right from `styles` rather than `styles/typeTokens` because it's nice to have easier imports and it means I (as a non-designer) don't have to know where these things are made under-the-hood.

### CODE REVIEW

File: src/styles/constants.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Allows lookup of font types without magic constants, eg:
+// `{ fontType: FONT_TYPES.body }` instead of `{ fontType: 'body' }`.
+export const FONT_TYPES = ['body', 'headings', 'interface'].reduce(
```
Comment (sarahmonster) [inline src/styles/constants.js]: Oh man, I'm going to have to learn how to use damned reducers, aren't I? 😭

File: src/styles/typeTokens.js [hunk 2]
```diff
@@ -1,108 +1,8 @@
...
-    fontFamily: theme.typography.fonts.interface,
-    fontWeight: theme.typography.fontWeights.interface,
-    ...typeAttributes({
-      theme,
-      sizeNumber: 1,
-      fontType: FONT_TYPES.interface,
-    }),
-  };
-};
+import { FONT_TYPES } from './constants';
```
Comment (sarahmonster) [inline src/styles/typeTokens.js]: Question: what is a font type here?

File: src/styles/typeTokens.js [hunk 3]
```diff
@@ -1,108 +1,8 @@
...
+import { smallCaps, typeAttributes } from './typeUtils';
```
Comment (sarahmonster) [inline src/styles/typeTokens.js]: ❤️

File: styleguide/components/Footer/index.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import React from 'react';
 import Version from 'rsg-components/Version';
 
 import theme from 'styleguide/theme';
-import { bodySmall, headingLarge } from 'themes/mixins';
+import { heading } from 'styles';
```
Comment (sarahmonster) [inline styleguide/components/Footer/index.js]: This is so much better. I wonder if just "styles" is too vague though? What about things like, say, focus styling? (Which will definitely be approached in a similar way to type styles here, and then applied to multiple components...)

File: styleguide/components/Footer/index.js [hunk 5]
```diff
@@ -15,7 +15,7 @@ export function Footer({ children, title, version }) {
         @media screen and (min-width: 960px) {
           grid-column: 1 / 3;
         }
-        ${css(headingLarge(theme))};
+        ${heading.large(theme)};
```
Comment (sarahmonster) [inline styleguide/components/Footer/index.js]: ❤️ So much nicer.

**Comment by sarahmonster** [comment]
Actually, my bad, checked out the wrong branch whilst trying to swap over to a more Atom-based workflow. 😅

**Comment by sarahmonster** [comment]
Also apparently there *was* a problem but I fixed it. Which may or may not have introduced some conflicts, my bad.

File: styleguide/components/Footer/index.js (hunk 4 above)
Comment (tofumatt) [reply styleguide/components/Footer/index.js]: You _can_ import from more specific folders/modules (`import { heading } from 'styles/typeTokens';` would work here). But I think having one consistent entrypoint is a nicer developer/designer experience. You can _store_ those focus styles in `styles/focus.js` (and should!), but you don't have to _import_ them that way in usage.

That's actually what's happening already here; check out `src/styles` to see how it works.

Comment (sarahmonster) [reply styleguide/components/Footer/index.js]: Ohhhhhhh I getcha. Coolio, let's do it your way.

File: src/styles/typeTokens.js (hunk 2 above)
Comment (tofumatt) [reply src/styles/typeTokens.js]: They're referencing values in themes, but it's all a bit hard-to-follow right now. I'm gonna see if I can improve it a bit...

**Comment by tofumatt** [comment]
The two issues you pointed out are fixed (one wasn't an issue, and one you fixed!)

I also rebased this onto the latest master and tidied things up a bit further. If you could have another look that'd be aces 😄

**Pull request merged**